keras_nlp/backend/config.py

```diff
@@ -65,10 +65,26 @@
 
 def detect_if_tensorflow_uses_keras_3():
     # We follow the version of keras that tensorflow is configured to use.
-    from tensorflow import keras
-
+    import keras
+    import tensorflow as tf
+    from packaging import version
+
+    # Return False if env variable is set and `tf_keras` is installed.
+    if os.environ.get("TF_USE_LEGACY_KERAS", None) in ("true", "True", "1"):
+        raise ValueError(
```

Comment:
Why are we ditching the `TF_USE_LEGACY_KERAS` support? With the previous version you had pushed here we would support it. No strong feelings on whether we support it or not, but not sure why we just flipped.